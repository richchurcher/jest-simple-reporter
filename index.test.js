import JestSimpleReporter from './'

describe('onTestResult', () => {
  const test = {}
  const results = {
    testFilePath: 'foo/bar/wombat',
    testResults: [{
      ancestorTitles: [ 'Flargle' ],
      failureMessages: [ `Error: expect(jest.fn()).not.toHaveBeenCalled()

Expected mock function not to be called but it was called with:
  Array []
    at Object.<anonymous> (${__dirname}/src/components/FlargleArgle/FlargleArgle.test.js:74:34)
    at Object.asyncFn (${__dirname}/node_modules/jest-jasmine2/build/jasmine-async.js:68:30)
    at resolve (${__dirname}/node_modules/jest-jasmine2/build/queueRunner.js:38:12)
    at tryCallTwo (${__dirname}/node_modules/promise/lib/core.js:45:1)
    at doResolve (${__dirname}/node_modules/promise/lib/core.js:200:9)
    at new Promise (${__dirname}/node_modules/promise/lib/core.js:66:1)
    at mapper (${__dirname}/node_modules/jest-jasmine2/build/queueRunner.js:31:21)
    at Promise.resolve.then.el (${__dirname}/node_modules/p-map/index.js:42:16)
    at tryCallOne (${__dirname}/node_modules/promise/lib/core.js:37:8)
    at ${__dirname}/node_modules/promise/lib/core.js:123:9` ],
      status: 'failed'
    }]
  }
  let log
  let reporter

  beforeEach(() => {
    log = jest.fn()
    global.console = { log }
    reporter = new JestSimpleReporter()
  })

  it('matches the last snapshot for a failing test', () => {
    reporter.onTestResult(test, results)
    expect(log.mock.calls[0][0]).toMatchSnapshot()
  })

  it('matches the last snapshot for a passsing test', () => {
    results.testResults = [{
      ancestorTitles: [ 'Flargle' ],
      failureMessages: [],
      status: 'passed',
      title: 'Works as expected'
    }]
    reporter.onTestResult(test, results)
    expect(log.mock.calls[0][0]).toMatchSnapshot()
  })
})
