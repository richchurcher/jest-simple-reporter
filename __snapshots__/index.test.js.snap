// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`onTestResult matches the last snapshot for a failing test 1`] = `
"
src/components/FlargleArgle/FlargleArgle.test.js:74:3:E:Expected mock function not to be called but it was called with: Array [] "
`;

exports[`onTestResult matches the last snapshot for a passsing test 1`] = `
"
foo/bar/wombat:::I:Flargle:Works as expected"
`;
